### PR TITLE
User can buy a subscription

### PR DESCRIPTION
**Pull request by erikbjoern**
# User story
[PT link](https://www.pivotaltracker.com/story/show/173033151)
```
As a user
In order to get access to the best articles
I would like to be able to buy a subscription
```

# Feature
* A user can become a subscriber in order to access premium content, by entering their card information to make a transaction
* For life time without possibility to cancel
* Error and success messages are displayed
* Request to back-end isn't made until a token could be successfully created by Stripe
* Visitors and users cannot see payment form until they are logged in

# Additions in this PR
* "Subscribe" tab in navbar
* `PaymentInterface`  - where the stripeToken is created through a direct call to Stripe, using the input from Stripe's own UI elements.
* `CreateSubscription` component, that holds `PaymentInterface` - this is where the stripeToken is used to make a call to our API which in turn communicates with Stripe (to make the charge and assign the customer to its subscription)

## Packages added:
* react-stripe-elements

# Screenshots
Screenshot 2020-05-30 at 16 30 28
Screenshot 2020-05-29 at 16 56 19
Screenshot 2020-05-29 at 16 57 50

### CODE REVIEW

**Review by jysmys — APPROVED** [review]

**Comment by ViaMarcus** [comment]
Checking the screen shots... why can you access the payment page without being logged in?
It should at least be locked saying log in first, but could display the prices when logged out

**Review by designerofthing — APPROVED** [review]

File: src/components/Navbar.jsx [hunk 1]
```diff
@@ -41,6 +41,11 @@ const Navbar = () => {
           </NavLink>
         </Menu.Item>
         {renderCategories}
+        <Menu.Item>
+          <NavLink to="/subscription" id='subscription-link'>
```
Comment (oliverochman) [inline src/components/Navbar.jsx:45]: when should a user see this link?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: src/components/Navbar.jsx (hunk 1 above)
Comment (erikbjoern) [reply src/components/Navbar.jsx:45]: When logged in, I changed it back to [WIP] for this

**Review by kermit-klein — APPROVED** [review]

**Review by ViaMarcus — APPROVED** [review]
Much better since the last few changes, good, easy-to-read code!

File: src/components/PaymentInterface.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      props.submitPayment(stripeResponse);
+    }
+  };
+
+  return (
+    <div className="payment-container">
+      <h4 className="error-message">{stripeTokenError}</h4>
+      <Segment padded inverted id="payment-interface">
+        <h3>Become a subscriber today!</h3>
+        <h4>Only $10 a month!</h4>
```
Comment (jysmys) [inline src/components/PaymentInterface.jsx:28]: Its a missmatch now with the subscriber fee....

**Review by oliverochman — APPROVED** [review]

**Pull request merged**